yt/data_objects/construction_data_containers.py

```diff
@@ -1404,8 +1415,10 @@ def _fill_fields(self, fields):
             )
             mylog.debug("Caught %d runtime errors.", runtime_errors_count)
         for name, v in zip(fields, ls.fields):
-            if self.level > 0:
-                v = v[1:-1, 1:-1, 1:-1]
+            # Remove buffer region
+            v = v[
+                self.nbuf : -self.nbuf, self.nbuf : -self.nbuf, self.nbuf : -self.nbuf
+            ]
```

Comment:
If you want to not repeat yourself, may I suggest
```suggestion
            # Remove buffer region
            sl = slice(self.nbuf, -self.nbuf)
            v = v[(sl, sl, sl)]
```

Reply:
Yes, I like this much better!